include/iam/publicnodeclient.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PUBLICNODECLIENT_HPP_
+#define PUBLICNODECLIENT_HPP_
+
+#include <atomic>
+#include <condition_variable>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <grpcpp/channel.h>
+#include <grpcpp/security/credentials.h>
+
+#include <aos/common/crypto.hpp>
+#include <aos/common/cryptoutils.hpp>
+#include <aos/common/tools/error.hpp>
+#include <aos/iam/certhandler.hpp>
+
+#include <iamanager/v5/iamanager.grpc.pb.h>
+
+#include "utils/time.hpp"
+
+namespace aos::common::iam {
+/**
+ * @brief Context for the crypto certificates.
+ */
+struct CryptoCertContext {
+    std::string                      mCertStorage;
+    std::string                      mCACert;
+    aos::cryptoutils::CertLoaderItf* certLoader;
+    aos::crypto::x509::ProviderItf*  cryptoProvider;
+};
+
+/**
+ * @brief Interface for the connection callback.
+ */
+class ConnectionCallbackItf {
+public:
+    virtual ~ConnectionCallbackItf() = default;
+
+    virtual bool OnConnected() = 0;
+};
+
+/**
+ * @brief Public node client interface.
+ */
+class PublicNodeClientItf : public ConnectionCallbackItf {
+public:
+    /**
+     * @brief Initialize the client.
```

Comment:
```suggestion
     * @brief Initializes the client.
```

Please fix the rest of brief comments